torch/ao/quantization/pt2e/prepare.py

```diff
@@ -71,7 +76,16 @@ def _maybe_insert_input_observer_for_arg_or_kwarg(
             assert isinstance(observed_arg, Node), f"expect observed argument to be a Node, but got: {type(observed_arg)}"
             assert observed_arg in obs_or_fq_map, \
                 f"can't refer to a node that does not have observer/fake_quant inserted yet: {observed_arg}"
-            arg_as_input_act_obs_or_fq = obs_or_fq_map[observed_arg]
+            if isinstance(input_arg_qspec, SharedQuantizationSpec):
```

Comment:
So if this is shared qspec, then wont you already be sharing observer? What is the observer replcaement being done here? Is it replacing the output observer of previous node? Not clear.

Reply:
yeah this is replacing the output observer from previous op

what do you mean by it's already sharing observer? user configures the input of the op to share with some other things, this is in the process of implementing it

Reply:
Basically if you have
conv1 (output=qint8).    conv2 (qint8)
  |                                        |
obs1                             obs2
 |                                     |
cat (wants shared obs)

then each conv1 and conv2 will already have their observers inserted when you come to cat. So I am asking will cat replcace obs1 with obs2? Or will you have some like

conv1 (output=qint8).    conv2 (qint8)
  |                                        |
obs1                             obs2
 |                                     |
 |                                     |
shared_obs3       shared_obs3
  |                             |
cat (wants shared obs)

It wasnt clear to me.

Reply:
my formatting is all messed up

Reply:
But basically if conv1 and conv2 have their observers inserted already and the output didnt have "shared quant spec", then here you are forcing it to share? That part wasnt clear to me.

Reply:
and then my question would be: is it better to do this as preprocess step on annotations, that figures out that conv1 and conv2's output is going into cat which wants shared qspec, then let me update conv1 and conv2's output_qspec to be shared?

Reply:
> But basically if conv1 and conv2 have their observers inserted already and the output didnt have "shared quant spec", then here you are forcing it to share? That part wasnt clear to me.

Yes

Reply:
> and then my question would be: is it better to do this as preprocess step on annotations, that figures out that conv1 and conv2's output is going into cat which wants shared qspec, then let me update conv1 and conv2's output_qspec to be shared?

hmmm, I think that would require additional analysis beforehand that would require more code, what is the benefit?

Reply:
I think both approaches are fine. was just pointing out alternatives.